### PR TITLE
Fix: date nil bug

### PR DESCRIPTION
**Pull request by aliiohs**
**What this PR does**:
#90 
**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: date.go [hunk 1]
```diff
@@ -26,6 +26,8 @@ import (
 // Date
 /////////////////////////////////////////
 
+var ZeroDate = time.Time{}
```
[the remark this answers is not in the document]
Comment (aliiohs) [reply date.go]: i think it need export in use.

[the remark this answers is not in the document]
Comment (aliiohs) [reply date.go]: i change it to exported by Method

[the remark this answers is not in the document]
Comment (aliiohs) [reply date.go]: I modified it to expose by method

File: date_test.go [hunk 2]
```diff
@@ -15,10 +15,25 @@
 package hessian
 
 import (
+	"github.com/stretchr/testify/assert"
```
Comment (wongoo) [inline date_test.go]: separate the import

File: date.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (wongoo) [reply date.go]: haven't changed， could be changed to `_zeroTime`

File: encode.go [hunk 3]
```diff
@@ -101,8 +101,15 @@ func (e *Encoder) Encode(v interface{}) error {
...
 
 	case time.Time:
+		if ZeroDate == val {
+			e.buffer = encNull(e.buffer)
+		} else {
+			e.buffer = encDateInMs(e.buffer, &val)
+			// e.buffer = encDateInMimute(v.(time.Time), e.buffer)
+		}
+
+	case *time.Time:
```
Comment (wongoo) [inline encode.go]: `*time.Time` not cover all situation, should move to `default` case and switch after `UnpackPtrType`

Comment (aliiohs) [reply encode.go]: v.(type)  is *time.Time and kind is struct ,if move default, i still deal with *time.Time

Comment (aliiohs) [reply encode.go]: What is the uncovered situation you said？

Comment (aliiohs) [reply encode.go]: already push my code, pls help me to review this code.

File: object.go [hunk 4]
```diff
@@ -424,12 +424,17 @@ func (d *Decoder) decInstance(typ reflect.Type, cls classInfo) (interface{}, err
...
 			)
-			if fldRawValue.Type().String() == "time.Time" {
+			typeName := fldRawValue.Type().String()
+			if strings.Contains(typeName, "time.Time") {
 				s, err = d.decDate(TAG_READ)
 				if err != nil {
 					return nil, perrors.WithStack(err)
 				}
-				fldRawValue.Set(reflect.ValueOf(s))
+				tValue := reflect.ValueOf(s)
```
Comment (wongoo) [inline object.go]: change to use func `SetValue` instead

**Comment by wongoo** [comment]
@aliiohs u can use func `SetValue` instead to set value, which will match the target and source type.

File: codec.go [hunk 5]
```diff
@@ -321,11 +331,6 @@ func SetValue(dest, v reflect.Value) {
 		v = UnpackPtrValue(v)
 	}
 
-	// zero value not need to set
-	if !v.IsValid() {
```
Comment (wongoo) [inline codec.go]: this can't be removed, the v may be unpacked and changed

**Comment by wongoo** [comment]
LGTM, great thanks to @aliiohs for the hard work~

**Pull request merged**